### PR TITLE
Display real plans in the deployment flow

### PR DESCRIPTION
**Pull request by huwshimi**
Display a list of real plans for the services in the deployment flow

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- use the df flag
- add a some charm with plans
- click deploy
- choose a cloud
- see the services, those with selected plans should display as such
- click change plan
- you should see the list of plans for that service

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5481/
Test PASSed.

File: jujugui/static/gui/src/app/components/budget-table/row/_row.scss [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
         margin-top: -11px;
     }
+
+    .expanding-row__initial {
+        height: auto;
+    }
+
+    .expanding-row__expanded {
+        margin-left: 0;
+    }
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/_row.scss]: The `__initial` and `__expanded` here are modifiers to the element. Therefore they should be `--initial` and `--expanded`

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/budget-table/row/_row.scss]: Ah, they're actually two different divs, one that will be displayed initially, one that will be displayed when the row is expanded.

Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/_row.scss]: Ah thanks.

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 2]
```diff
@@ -24,17 +24,60 @@ YUI.add('budget-table-row', function() {
...
+        this.plansXHR = this.props.listPlansForCharm(
+          this.props.service.get('charm'), this._getPlansCallback);
+      });
+    },
+
+    /**
+      Callback for when plans for an entity have been successfully fetched.
+      @method _getPlansCallback
+      @param {String} error An error message, or null if there's no error.
+      @param {Array} models A list of the plans found.
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: plans

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 3]
```diff
@@ -167,40 +210,70 @@ YUI.add('budget-table-row', function() {
...
+            -
+          </span>);
+      }
+    },
+
     render: function() {
       var plansEditable = this.props.plansEditable;
       var classes = {
-        'budget-table__row': true,
+        'budget-table-row': true,
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Is `budget-table-row` now a component?

**Comment by anthonydillon** [comment]
Code is good with a few comments :+1:

**Comment by anthonydillon** [comment]
Following the QA steps. 
 - I added landscape-mock which has plans 
 - I also added Kabana which doesn't have plans
 - Went to the DF and selected a cloud
 - Scrolled down to the Services and plans section and both were listed (image below)
 - I didn't expect to see Kabana in the list as it doesn't have plans?

default juju gui

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 4]
```diff
@@ -97,7 +140,7 @@ YUI.add('budget-table-row', function() {
      @returns {Object} The plan form.
     */
     _generateChangePlan: function() {
-      if (!this.props.plansEditable) {
+      if (!this.props.plansEditable || this.state.plans.length === 0) {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Is it possible that 'this.state.plans' could be null? We've been bitten by this elsewhere, so a little type safety check might be good here.

**Comment by kadams54** [comment]
:+1: with one minor comment. Thanks!

File: jujugui/static/gui/src/app/components/budget-table/row/row.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: It sure is!

**Comment by huwshimi** [comment]
@anthonydillon from what I can tell from the mockups if a service doesn't need a plan it gets listed but under details it just lists a "-" and shouldn't have the usage/allocation/spend details, but they appear currently as that data is mocked.

File: jujugui/static/gui/src/app/components/budget-table/row/row.js (hunk 4 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Ah yep, good call.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5492/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @anthonydillon and @kadams54 :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**